clang/test/Preprocessor/has_attribute.cpp

```diff
@@ -55,7 +55,7 @@ CXX11(unlikely)
 // CHECK: likely: 201803L
 // CHECK: maybe_unused: 201603L
 // ITANIUM: no_unique_address: 201803L
-// WINDOWS: no_unique_address: 0
+// WINDOWS: no_unique_address: 201803L 
```

Comment:
Correct, this appears to be a bug (possibly with the way we're generating code from ClangAttrEmitter.cpp); this should be `0` on `WINDOWS`. I think this is happening because we removed `TargetSpecificAttr` from the attribute definition -- if I'm correct, we may require some deeper changes to express target-specific *spellings* of an attribute instead of making the attribute as a whole target-specific. CC @erichkeane as there's possibly going to be some design questions there.

Reply:
OOoh, actually, here's an idea that *might* work without needing to do tablegen surgery:
```
def NoUniqueAddress : InheritableAttr {
  let Subjects = SubjectList<[NonBitField], ErrorDiag>;
  let Documentation = [NoUniqueAddressDocs];
}

def MSNoUniqueAddress : NoUniqueAddress, TargetSpecificAttr<TargetMicrosoftCXXABI> {
  let Spellings = [CXX11<"msvc", "no_unique_address", 201803>];
}

def ItaniumNoUniqueAddress : NoUniqueAddress, TargetSpecificAttr<TargetItaniumCXXABI> {
  let Spellings = [CXX11<"", "no_unique_address", 201803>];
}
```
We'd have to do `let AdditionalMembers = ` in the base class to manually implement what is done automatically with `let Accessors =`, but this might work. Worth noting: totally untested idea, so it could also explode horribly.

Reply:
Tried this and the inheritance part wasn't quite working (think that the NoUniqueAddress should be a class, but in that case, not sure how to parse the two attributes). I did a slightly different version where we have three attributes with no inheritance (MS, Itanium, and general) and just parse the MS and Itanium versions as the general version in SemaDeclAttr. It at least passes the tests that were broken before.